src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,144 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String INDEX_EXCEPTION = "Index is incorrect";
+    private Node<T> node;
+    private Node<T> tail;
+    private Node<T> head;
+    private int size;
+
     @Override
     public void add(T value) {
+        node = new Node<>(null, value, null);
+        if (head == null) {
+            head = node;
+        }
+        node.prev = tail;
+        if (tail != null) {
+            tail.next = node;
+        }
+        tail = node;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException(INDEX_EXCEPTION);
+        }
+        Node<T> newNode = new Node<>(null, value, null);
+        node = findByIndex(index);
+        if (node == null) {
+            add(value);
+            return;
+        } else if (node.prev == null) {
+            head.prev = new Node<>(null, value, head);
+            head = head.prev;
+        } else {
+            newNode.next = node;
+            newNode.prev = node.prev;
+            newNode.prev.next = newNode;
+            newNode.next.prev = newNode;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkPositionIndex(index);
+        return findByIndex(index).value;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkPositionIndex(index);
+        T oldValue = findByIndex(index).value;
+        node = findByIndex(index);
+        node.value = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkPositionIndex(index);
+        T oldValue = findByIndex(index).value;
+        node = findByIndex(index);
+        unlink(node);
+        return oldValue;
```

Comment:
i think you can just return node.value
and delete T old value = findByIndex(index).value